File: codeforces/low_difficulty/difficulty_1000/8.dragons_[230A]/dragons.py
# Author: José Rodolfo (jric2002)
s, n = map(int, input().split(" "))
d = []
status = True
while (n):
  d.append([int (x) for x in input().split(" ")])
  n -= 1
d.sort(key=lambda x : x[0])
for p in d:
  if (s <= p[0]):
    status = False
    break
  else:
    s += p[1]
print("YES") if (status) else print("NO")

File: codeforces/low_difficulty/difficulty_1000/8.dragons_[230A]/dragons.cpp
#include <iostream>
#include <utility>
#include <vector>
#include <algorithm>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  int s, n, x, y;
  vector<pair<int, int>> d;
  bool status;
  cin >> s >> n;
  status = true;
  for (int i = 0; i < n; i++) {
    cin >> x >> y;
    d.push_back(make_pair(x, y));
  }
  sort(d.begin(), d.end());
  for (pair<int, int> p : d) {
    if (s <= p.first) {
      status = false;
      break;
    }
    else {
      s += p.second;
    }
  }
  (status) ? cout << "YES" << endl : cout << "NO" << endl;
  return 0;
}